src/main/java/de/diddiz/LogBlock/BlockChange.java

```diff
@@ -150,7 +152,7 @@ public BaseComponent[] getLogMessage(int entry) {
                     msg.addExtra(createTextComponentWithColor(" into ", CREATE.getColor()));
                     msg.addExtra(prettyMaterial(type));
                 }
-            } else if (type instanceof Waterlogged && ((Waterlogged) type).isWaterlogged() != ((Waterlogged) replaced).isWaterlogged()) {
+            } else if (isWaterlogged(type, (Waterlogged) replaced)) {

```

Comment:
this won't even work. it is not guaranteed that replaced is Waterlogged. And it definitely decreases readability